packages/comms/src/services/wsDali.ts

```diff
@@ -4,5 +4,36 @@ export {
     WsDali
 };
 
+export interface ExportRequest {
+    Path: string;
+    Safe: boolean;
+}
+
 export class DaliService extends DaliServiceBase {
+
+    async Export(request: ExportRequest): Promise<void> {
+        const exportUrl = "/WsDali/Export";
+
+        const reqParams = new URLSearchParams();
+        if (request.Path) {
+            reqParams.append("Path", request.Path);
+        }
+        if (request.Safe !== undefined) {
+            reqParams.append("Safe", String(request.Safe));
+        }
+
+        const response = await fetch(`${exportUrl}?${reqParams.toString()}`);
+        const filename = response.headers.get("Content-Disposition");
+        const blob = await response.blob();
+
+        const urlObj = window.URL.createObjectURL(blob);
+
+        const link = document.createElement("a");
+        link.href = urlObj;
+        link.download = filename ? filename.split("filename=")[1] : "WsDali_Export";
+        link.click();
+        link.remove();
+        window.URL.revokeObjectURL(urlObj);
```

Comment:
You can't assume that `document` and `window` will be available (NodeJS, ECL Extension etc.).  It should just return the blob.

Alternatively you could just (in ECL Watch):
```
        const link = document.createElement("a");
        link.href = `${exportUrl}?${reqParams.toString()}`;
        link.download = filename ? filename.split("filename=")[1] : "WsDali_Export";
        link.click();
        link.remove();
        window.URL.revokeObjectURL(urlObj);
  ```